compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -498,13 +539,13 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 {
                     let _ = path.segments.pop();
                     if rename.is_none() {
-                        *ident = path.segments.last().unwrap().ident;
+                        ident = path.segments.last().unwrap().ident;
                     }
                 }
 
                 let res = self.lower_import_res(id, path.span);
                 let path = self.lower_use_path(res, &path, ParamMode::Explicit);
-                hir::ItemKind::Use(path, hir::UseKind::Single)
+                hir::ItemKind::Use(path, hir::UseKind::Single(ident))
```

Comment:
Unlowered ident passed to `Single`

```suggestion
                let ident = self.lower_ident(ident);
                hir::ItemKind::Use(path, hir::UseKind::Single(ident))
```

Reply:
Good catch!